disnake/gateway.py

```diff
@@ -199,9 +199,9 @@ def run(self) -> None:
                     f.result()
                 except Exception:
                     _log.exception("An error occurred while stopping the gateway. Ignoring.")
+                    return
                 finally:
                     self.stop()
-                    return  # noqa: B012
```

Comment:
This also changes behavior, continuing to heartbeat after closing the ws.
As noted in the PEP and B012, `return` in `finally` can swallow exceptions (in this case only `BaseException`s), which I would argue is fine here; the thread is at the end of its lifecycle anyway.